app/main.py

```diff
@@ -1,8 +1,21 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list) -> list:
+    for person_dict in people_list:
+        Person(person_dict["name"], person_dict["age"])
+
+    for person_dict in people_list:
+        person = Person.people[person_dict["name"]]
+        if "wife" in person_dict and person_dict["wife"]:
+            person.wife = Person.people[person_dict["wife"]]
```

Comment:
You could simplify this code and make it more efficient using walrus
```suggestion
        if wife_name := person_dict.get("wife"):
            person.wife = Person.people[wife_name]
```